src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -314,6 +403,194 @@ private void ResolveAsymmetricAlgorithm(AsymmetricSecurityKey key, string algori
+
+            if (webKey.X == null)
+                throw LogHelper.LogArgumentNullException(nameof(webKey.X));
+
+            if (webKey.Y == null)
+                throw LogHelper.LogArgumentNullException(nameof(webKey.Y));
+
+            GCHandle keyBlobHandle = new GCHandle();
+            try
+            {
+                uint dwMagic = GetMagicValue(webKey.Crv, willCreateSignatures);
+                uint cbKey = GetKeyByteCount(webKey.Crv);
+                byte[] keyBlob;
+                if (willCreateSignatures)
+                    keyBlob = new byte[3 * cbKey + 2 * Marshal.SizeOf<uint>()];
+                else
+                    keyBlob = new byte[2 * cbKey + 2 * Marshal.SizeOf<uint>()];
+
+                keyBlobHandle = GCHandle.Alloc(keyBlob, GCHandleType.Pinned);
+                IntPtr keyBlobPtr = keyBlobHandle.AddrOfPinnedObject();
+                byte[] x = Base64UrlEncoder.DecodeBytes(webKey.X);
+                byte[] y = Base64UrlEncoder.DecodeBytes(webKey.Y);
+
+                Marshal.WriteInt64(keyBlobPtr, 0, dwMagic);
+                Marshal.WriteInt64(keyBlobPtr, 4, cbKey);
+
+                int index = 8;
+                foreach (byte b in x)
+                    Marshal.WriteByte(keyBlobPtr, index++, b);
+
+                foreach (byte b in y)
+                    Marshal.WriteByte(keyBlobPtr, index++, b);
+
+                if (willCreateSignatures)
+                {
+                    if (webKey.D == null)
+                        throw LogHelper.LogArgumentNullException(nameof(webKey.D));
+
+                    byte[] d = Base64UrlEncoder.DecodeBytes(webKey.D);
+                    foreach (byte b in d)
+                        Marshal.WriteByte(keyBlobPtr, index++, b);
+
+                    Marshal.Copy(keyBlobPtr, keyBlob, 0, keyBlob.Length);
+                    using (CngKey cngKey = CngKey.Import(keyBlob, CngKeyBlobFormat.EccPrivateBlob))
+                    {
+                        _ecdsa = new ECDsaCng(cngKey);
+                    }
+                }
+                else
+                {
+                    Marshal.Copy(keyBlobPtr, keyBlob, 0, keyBlob.Length);
+                    using (CngKey cngKey = CngKey.Import(keyBlob, CngKeyBlobFormat.EccPublicBlob))
+                    {
+                        _ecdsa = new ECDsaCng(cngKey);
+                    }
+                }
+                keyBlobHandle.Free();
+            }
+            finally
+            {
+                if (keyBlobHandle != null)
+                    keyBlobHandle.Free();
+            }
+        }
+
+        /// <summary>
+        /// Returns the size of key in bytes
+        /// </summary>
+        /// <param name="curveId">represents ecdsa curve -P256, P384, P512</param>
+        /// <returns>size of the key in bytes</returns>
+        private uint GetKeyByteCount(string curveId)
+        {
+            if (string.IsNullOrEmpty(curveId))
```

Comment:
Why remove this if we want to caution future developers that is not checked for null?